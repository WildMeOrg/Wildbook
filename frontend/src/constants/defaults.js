export const defaultAreaBounds = {
    north: 16.892664506753874,
    south: -11.91322015697743,
    east: 14.857142857142884,
    west: -15.428571428571413,
  };
  
  export const defaultCrossfadeDuration = 500;
  