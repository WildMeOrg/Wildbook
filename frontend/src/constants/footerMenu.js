const footerLinks1 = [
  {
    href: "https://secure.givelively.org/donate/conservation-x-labs-inc/wild-me",
    id: "FOOTER_DONATE",
  },
  {
    href: "https://wildbook.docs.wildme.org/getting-started-with-wildbook.html",
    id: "FOOTER_DOCUMENTATION",
  },
];
const footerLinks2 = [
  { href: "https://community.wildme.org", id: "FOOTER_COMMUNITY_FORUM" },
  { href: "https://github.com/WildMeOrg", id: "FOOTER_GITHUB" },
];
const footerLinks3 = [
  {
    href: "https://www.instagram.com/conservationxlabs",
    id: "FOOTER_INSTAGRAM",
  },
  { href: "https://www.facebook.com/ConservationXLabs", id: "FOOTER_FACEBOOK" },
  { href: "https://twitter.com/conservationx", id: "FOOTER_TWITTER" },
  {
    href: "https://www.linkedin.com/company/conservationxlabs/",
    id: "FOOTER_LINKEDIN",
  },
];

export { footerLinks1, footerLinks2, footerLinks3 };
