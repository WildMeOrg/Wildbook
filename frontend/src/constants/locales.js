const locales = ["en", "es", "fr", "it", "de"];
const localeMap = {  
  en: "en",
  es: "es",
  fr: "fr",
  it: "it",
  de: "de",
};

const languageMap = {  
  en: "English",
  es: "Español",
  fr: "Français",
  it: "Italiano",
  de: "Deutsch",
};

export { locales, localeMap, languageMap };
