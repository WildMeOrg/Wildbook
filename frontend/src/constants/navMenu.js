import React from "react";
import { FormattedMessage } from "react-intl";

const authenticatedMenu = (username) => [
  {
    Submit: [
      {
        name: (
          <FormattedMessage
            id="menu.submit.reportSighting"
            defaultMessage="Report an Encounter"
          />
        ),
        href: "/submit.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.submit.bulkImport"
            defaultMessage="Bulk Import"
          />
        ),
        href: "/import/instructions.jsp",
      },
    ],
  },
  {
    Learn: [
      {
        name: (
          <FormattedMessage
            id="menu.learn.aboutWildbook"
            defaultMessage="About Wildbook"
          />
        ),
        href: "/overview.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.contactUs"
            defaultMessage="Contact Us"
          />
        ),
        href: "/contactus.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.citingWildbook"
            defaultMessage="Citing Wildbook"
          />
        ),
        href: "/react/citation",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.howToPhotograph"
            defaultMessage="How to Photograph"
          />
        ),
        href: "/photographing.jsp",
      },
      // { name: <FormattedMessage id="menu.learn.privacyPolicy" defaultMessage="Privacy Policy" />, href: '/privacyPolicy.jsp' },
      // { name: <FormattedMessage id="menu.learn.termsOfUse" defaultMessage="Terms of Use" />, href: '/termsOfUse.jsp' },
    ],
  },

  {
    My_Data: [
      {
        name: (
          <FormattedMessage
            id="menu.myData.myEncounters"
            defaultMessage="My Encounters"
          />
        ),
        href: `/react/encounter-search?username=${username}`,
        sub: [
          {
            name: (
              <FormattedMessage
                id="menu.myData.approvedAnimals"
                defaultMessage="My Approved Animals"
              />
            ),
            href: `/react/encounter-search?username=${username}&state=approved`,
          },
          {
            name: (
              <FormattedMessage
                id="menu.myData.unapprovedAnimals"
                defaultMessage="My Unapproved Animals"
              />
            ),
            href: `/react/encounter-search?username=${username}&state=unapproved`,
          },
          {
            name: (
              <FormattedMessage
                id="menu.myData.unidentifiableAnimals"
                defaultMessage="My Unidentifiable Animals"
              />
            ),
            href: `/react/encounter-search?username=${username}&state=unidentifiable`,
          },
        ],
      },
      {
        name: (
          <FormattedMessage
            id="menu.myData.myIndividuals"
            defaultMessage="My Individuals"
          />
        ),
        href: `/individualSearchResults.jsp?username=${username}`,
      },
      {
        name: (
          <FormattedMessage
            id="menu.myData.mySightings"
            defaultMessage="My Sightings"
          />
        ),
        href: `/occurrenceSearchResults.jsp?submitterID=${username}`,
      },
      {
        name: (
          <FormattedMessage
            id="menu.myData.myBulkImports"
            defaultMessage="My Bulk Imports"
          />
        ),
        href: "/imports.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.myData.myProjects"
            defaultMessage="My Projects"
          />
        ),
        href: "/projects/projectList.jsp",
      },
    ],
  },
  {
    Search: [
      {
        name: (
          <FormattedMessage
            id="menu.search.encounters"
            defaultMessage="Encounters"
          />
        ),
        href: "/react/encounter-search",
      },
      {
        name: (
          <FormattedMessage
            id="menu.search.individuals"
            defaultMessage="Individuals"
          />
        ),
        href: "/individualSearch.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.search.sightings"
            defaultMessage="Sightings"
          />
        ),
        href: "/occurrenceSearch.jsp",
      },
    ],
  },
  {
    Animals: [
      {
        name: (
          <FormattedMessage
            id="menu.animals.individualGallery"
            defaultMessage="Individual Gallery"
          />
        ),
        href: "/gallery.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.animals.animalCalendar"
            defaultMessage="Animal Calendar"
          />
        ),
        href: "/xcalendar/calendar.jsp",
      },
    ],
  },
  {
    Administer: [
      {
        name: (
          <FormattedMessage
            id="menu.administer.manageAccounts"
            defaultMessage="Manage My Accounts"
          />
        ),
        href: "/myUsers.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.userManagement"
            defaultMessage="User Management"
          />
        ),
        href: "/appadmin/users.jsp?context=context0",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.libraryAdmin"
            defaultMessage="Library Administration"
          />
        ),
        href: "/appadmin/admin.jsp",
      },
      {
        name: (
          <FormattedMessage id="menu.administer.logs" defaultMessage="Logs" />
        ),
        href: "/appadmin/logs.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.photoKeywords"
            defaultMessage="Photo Keywords"
          />
        ),
        href: "/appadmin/kwAdmin.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.softwareDocs"
            defaultMessage="Software Documentation"
          />
        ),
        href: "https://wildbook.docs.wildme.org/introduction/index.html",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.dataIntegrity"
            defaultMessage="Data Integrity"
          />
        ),
        href: "/appadmin/dataIntegrity.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.administer.bulkImportLogs"
            defaultMessage="Bulk Import Logs"
          />
        ),
        href: "/imports.jsp",
      },
    ],
  },
];

const unAuthenticatedMenu = [
  {
    Submit: [
      {
        name: (
          <FormattedMessage
            id="menu.submit.reportSighting"
            defaultMessage="Report an Encounter"
          />
        ),
        href: "/submit.jsp",
      },
    ],
  },
  {
    Learn: [
      {
        name: (
          <FormattedMessage
            id="menu.learn.aboutWildbook"
            defaultMessage="About Wildbook"
          />
        ),
        href: "/overview.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.contactUs"
            defaultMessage="Contact Us"
          />
        ),
        href: "/photographing.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.citingWildbook"
            defaultMessage="Citing Wildbook"
          />
        ),
        href: "/react/citation",
      },
      {
        name: (
          <FormattedMessage
            id="menu.learn.howToPhotograph"
            defaultMessage="How to Photograph"
          />
        ),
        href: "/photographing.jsp",
      },
      //   {
      //     name: (
      //       <FormattedMessage
      //         id="menu.learn.privacyPolicy"
      //         defaultMessage="Privacy Policy"
      //       />
      //     ),
      //     href: "/privacy-policy.jsp",
      //   },
      //   {
      //     name: (
      //       <FormattedMessage
      //         id="menu.learn.termsOfUse"
      //         defaultMessage="Terms of Use"
      //       />
      //     ),
      //     href: "/terms-of-use.jsp",
      //   },
    ],
  },

  {
    Animals: [
      {
        name: (
          <FormattedMessage
            id="menu.animals.individualGallery"
            defaultMessage="Individual Gallery"
          />
        ),
        href: "/gallery.jsp",
      },
      {
        name: (
          <FormattedMessage
            id="menu.animals.animalCalendar"
            defaultMessage="Animal Calendar"
          />
        ),
        href: "/xcalendar/calendar.jsp",
      },
    ],
  },
];

export { authenticatedMenu, unAuthenticatedMenu };
