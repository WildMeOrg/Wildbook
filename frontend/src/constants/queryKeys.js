function sortIfArray(value) {
    return Array.isArray(value) ? value.sort() : value;
  }
  
  export default {
    me: 'me',
    settingsSchema: 'settingsSchema',
    siteInfo: 'siteInfo',
    homePageInfo: 'homePageInfo',
    notifications: 'notifications',
    detectionConfig: 'detectionConfig',
    keywords: 'keywords',
    users: 'users',
    collaborations: 'collaborations',
    assetGroupSightings: 'assetGroupSightings',
    allNotifications: 'allNotifications',
    unreadNotifications: 'unreadNotifications',
    twitterBotTestResults: 'twitterBotTestResults',
    publicData: 'publicData',
    socialGroups: 'socialGroups',
    publicAssetGroupSightings: 'publicAssetGroupSightings',
    sageJobs: ['sage', 'jobs'],
  };
  
  export function getEncounterQueryKey(guid) {
    return ['encounter', guid];
  }
  
  export function getSightingQueryKey(guid) {
    return ['sighting', guid];
  }
  
  export function getSightingMatchResultsQueryKey(guid) {
    return ['sightingMatchResults', guid];
  }
  
  export function getAGSQueryKey(guid) {
    return ['assetGroupSighting', guid];
  }
  
  export function getIndividualQueryKey(guid) {
    return ['individual', guid];
  }
  
  export function getIndividualMergeRequestQueryKey(guid) {
    return ['individualMergeRequest', guid];
  }
  
  export function getMergeConflictsQueryKey(guids) {
    return ['individualMergeConflicts', guids];
  }
  
  export function getUserQueryKey(guid) {
    return ['user', guid];
  }
  
  export function getUserSightingsQueryKey(guid, query, params) {
    return ['userSightings', guid, query, params];
  }
  
  export function getUserAgsQueryKey(guid, params) {
    return ['userAgs', guid, params];
  }
  
  export function getAssetGroupQueryKey(guid) {
    return ['assetGroup', guid];
  }
  
  export function getNotificationQueryKey(guid) {
    return ['notification', guid];
  }
  
  export function getIndividualTermQueryKey(searchTerm) {
    return ['individualQuickSearch', searchTerm];
  }
  
  export function getIndividualsByGuidsQueryKey(individualGuids) {
    return ['individualQuickSearch', sortIfArray(individualGuids)];
  }
  
  export function getSightingTermQueryKey(searchTerm) {
    return ['sightingQuickSearch', searchTerm];
  }
  
  export function getEncounterTermQueryKey(searchTerm) {
    return ['encounterQuickSearch', searchTerm];
  }
  
  export function getSocialGroupQueryKey(guid) {
    return ['socialGroup', guid];
  }
  
  export function getIndividualFilterQueryKey(
    filters,
    page,
    rowsPerPage,
  ) {
    return ['individualFilterSearch', filters, page, rowsPerPage];
  }
  
  export function getSightingFilterQueryKey(
    filters,
    page,
    rowsPerPage,
  ) {
    return ['sightingFilterSearch', filters, page, rowsPerPage];
  }
  
  export function getEncounterFilterQueryKey(
    filters,
    page,
    rowsPerPage,
  ) {
    return ['encounterFilterSearch', filters, page, rowsPerPage];
  }
  
  export function getAuditLogQueryKey(filters, page, rowsPerPage) {
    return ['auditLogFilterSearch', filters, page, rowsPerPage];
  }
  