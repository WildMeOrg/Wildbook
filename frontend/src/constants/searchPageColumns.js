export const encounterSearchColumns = [
  { name: "INDIVIDUAL_ID", selector: "individualDisplayName" },
  { name: "SIGHTING_ID", selector: "occurrenceId" },
  { name: "ALTERNATIVE_ID", selector: "otherCatalogNumbers" },
  { name: "CREATED_DATE", selector: "date" },
  { name: "LOCATION_ID", selector: "locationId" },
  { name: "SPECIES", selector: "taxonomy" },
  { name: "SUBMITTER", selector: "assignedUsername" },
  { name: "DATE_SUBMITTED", selector: "dateSubmitted" },
  { name: "NUMBER_ANNOTATIONS", selector: "numberAnnotations" },
];
