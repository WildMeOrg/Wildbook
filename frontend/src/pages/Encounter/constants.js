
const SECTION_FIELD_PATHS = {
  date: ["date", "verbatimEventDate"],
  identify: [
    "individualDisplayName",
    "otherCatalogNumbers",
    "identificationRemarks",
    "occurrenceId",
    "sightingId",
    "individualId",
  ],
  metadata: [
    "id",
    "submitterID",
    "state",
    "observationComments",
  ],
  location: [
    "verbatimLocality",
    "locationId",
    "locationName",
    "country",
    "locationGeoPoint",
  ],
  attributes: [
    "taxonomy",
    "livingStatus",
    "sex",
    "distinguishingScar",
    "behavior",
    "groupRole",
    "patterningCode",
    "lifeStage",
    "occurrenceRemarks",
  ],
};

const LOCAL_FIELD_ERRORS = {
  date: {
    date: "invalid date"
  },
  location: {
    latitude: "please enter invalid latitude and longitude",
    longitude: "please enter invalid latitude and longitude",
  },
  measurements: {
    measurements: "please enter valid values",
  }
};

export { SECTION_FIELD_PATHS, LOCAL_FIELD_ERRORS };