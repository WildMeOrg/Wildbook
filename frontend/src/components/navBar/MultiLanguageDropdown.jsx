import React, { useContext, useState } from "react";
import Dropdown from "react-bootstrap/Dropdown";
import DownIcon from "../svg/DownIcon";
import LocaleContext from "../../IntlProvider";
import { locales, localeMap, languageMap } from "../../constants/locales";
import Cookies from "js-cookie";

export default function MultiLanguageDropdown() {
  const { onLocaleChange } = useContext(LocaleContext);
  const initialLocale = Cookies.get("wildbookLangCode") || "en";
  const [flag, setFlag] = useState(initialLocale);
  return (
    <div
      className="d-flex align-items-center justify-content-center border-0 rounded-pill m-2"
      style={{
        backgroundColor: "rgba(255, 255, 255, 0.25)",
        width: "65px",
        height: "35px",
      }}
    >
      <Dropdown>
        <Dropdown.Toggle variant="basic" id="dropdown-basic">
          <img
            src={`${process.env.PUBLIC_URL}/flags/${flag}.png`}
            alt="flag"
            style={{ width: "20px", height: "12px" }}
          />
          <span style={{ paddingLeft: 7 }}>
            <DownIcon />
          </span>
        </Dropdown.Toggle>

        <Dropdown.Menu>
          {locales.map((locale, index) => (
            <Dropdown.Item
              key={index}
              onClick={() => {
                onLocaleChange(locale);
                setFlag(localeMap[locale]);
              }}
            >
              <img
                src={`${process.env.PUBLIC_URL}/flags/${localeMap[locale]}.png`}
                alt={locale}
                style={{ width: "20px", height: "12px", marginRight: "10px" }}
              />
              {languageMap[locale]}
            </Dropdown.Item>
          ))}
        </Dropdown.Menu>
      </Dropdown>
    </div>
  );
}
