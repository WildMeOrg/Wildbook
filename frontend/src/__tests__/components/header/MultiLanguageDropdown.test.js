import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import MultiLanguageDropdown from "../../../components/navBar/MultiLanguageDropdown";
import LocaleContext from "../../../IntlProvider";
import { localeMap, languageMap } from "../../../constants/locales";
import "@testing-library/jest-dom";
import Cookies from "js-cookie";

beforeAll(() => {
  process.env.PUBLIC_URL = "/react";
});

jest.mock("js-cookie", () => ({
  get: jest.fn(),
}));

const mockOnLocaleChange = jest.fn();
Cookies.get.mockReturnValue("en");

test("renders MultiLanguageDropdown and changes language on click", async () => {

  const { getByAltText } = render(
    <LocaleContext.Provider value={{ onLocaleChange: mockOnLocaleChange }}>
      <MultiLanguageDropdown />
    </LocaleContext.Provider>,
  );

  expect(getByAltText("flag")).toBeInTheDocument();
  expect(getByAltText("flag").src).toContain("/react/flags/en.png");

});

test("renders changes language on click", async () => {
  const { getByRole, getByText, getByAltText } = render(
    <LocaleContext.Provider value={{ onLocaleChange: mockOnLocaleChange }}>
      <MultiLanguageDropdown />
    </LocaleContext.Provider>,
  );

  await act(async () => {
    fireEvent.click(getByRole("button"));
  });

  await act(async () => {
    fireEvent.click(getByText(languageMap["fr"]));
  });

  expect(getByAltText("fr").src).toContain(
    `/react/flags/${localeMap["fr"]}.png`,
  );
  expect(mockOnLocaleChange).toHaveBeenCalledWith("fr");
});