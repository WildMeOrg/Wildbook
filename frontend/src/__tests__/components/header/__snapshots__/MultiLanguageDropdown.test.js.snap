// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MultiLanguageDropdown and changes language on click 1`] = `
<a
  class="dropdown-item"
  data-rr-ui-dropdown-item=""
  href="#"
  role="button"
  tabindex="0"
>
  <img
    alt="fr"
    src="/react/flags/fr.png"
    style="width: 20px; height: 12px; margin-right: 10px;"
  />
  Fran√ßais
</a>
`;
